pyomo/contrib/gdpopt/branch_and_bound.py

```diff
@@ -37,442 +46,483 @@
-
-    term_cond = result.solver.termination_condition
-    if term_cond == tc.optimal:
-        assert result.solver.status is SolverStatus.ok
-        lb = result.problem.lower_bound if not obj_sense_correction else \
-             -result.problem.upper_bound
-        ub = result.problem.upper_bound if not obj_sense_correction else \
-             -result.problem.lower_bound
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config,
-        )
-        return float('-inf'), ub
-    elif term_cond == tc.locallyOptimal or term_cond == tc.feasible:
-        assert result.solver.status is SolverStatus.ok
-        lb = result.problem.lower_bound if not obj_sense_correction else \
-             -result.problem.upper_bound
-        ub = result.problem.upper_bound if not obj_sense_correction else \
-             -result.problem.lower_bound
-        # TODO handle LB absent
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config,
-        )
-        return float('-inf'), ub
-    elif term_cond == tc.unbounded:
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config, ignore_integrality=True
-        )
-        return float('-inf'), float('-inf')
-    elif term_cond == tc.infeasible:
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config, ignore_integrality=True
-        )
-        return float('-inf'), float('inf')
-    else:
-        config.logger.warning("Unknown termination condition of %s. "
-                              "Treating as infeasible." % term_cond)
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config, ignore_integrality=True
-        )
-        return float('-inf'), float('inf')
+            config.logger.warning("Unknown termination condition of %s. "
+                                  "Treating as infeasible." % term_cond)
+            copy_var_list_values(
+                from_list=subprob_utils.algebraic_variable_list,
+                to_list=model_utils.algebraic_variable_list,
+                config=config, ignore_integrality=True
+            )
+            return float('inf'), float('inf')
+
+
+    def _solve_local_rnGDP_subproblem(self, model, config):
+        # TODO: The returns of this method should be improved. Currently, it
+        # returns trivial bounds (LB, UB) = (-inf, inf) if there is an error in
+        # the solve, if the problem is infeasible, or if the problem is
+        # unbounded.
```

Comment:
Is there anything better we could do in these cases? I think we could do better in the infeasible case such that the branch is "closed" but besides that, the unbounded and error behaviors is what I would expect.

Reply:
It's apparently not tested... I just changed the infeasible return to be `float('inf'), float('inf')` and nothing broke. I agree with you that that seems more reasonable (since the problem is always corrected to be a minimization). But I'm a little nervous to changes here without taking the time to figure out why it was this way in the first place. Which is an exercise I would really like to defer to a future PR.

Reply:
Perfect. Maybe then add a comment for future developers or an issue here in GitHub for us to remember.